tasks/hostname.yml

```diff
@@ -18,6 +18,6 @@
     mh_hostname is defined
     or (
       mh_current_hostname.stdout is defined
-      and mh_current_hostname.stdout != mh_current_hostname
+      and mh_current_hostname.stdout != mh_hostname
```

Comment:
I don't think this change is quite right. The code it's replacing *is* wrong, but this change makes it possible to accidentally run this command on the box:

```
scutil --set Hostname ''
```

I think what's wanted is more along the lines of the following, as it prevents setting the hostname to an empty string, and the command won't run _at all_ when `mh_hostname` and `mh_current_hostname.stdout` are identical.

As near as I can remember, that was the original intent.

```yml
  when:
    - "mh_hostname is defined"
    - "mh_hostname | length > 0"
    - "mh_current_hostname.stdout != mh_hostname"
```

Reply:
Makes sense to me.

Personally I tend to define most input variables as defaults (to some empty values) so I don't need to handle "undefined", just the empty values.

Reply:
This role actually does that too. It's belt and suspenders around here :grin:

Reply:
Setting Hostname to "" actually unset the HostName 
```
scutil --set HostName ""
scutil --get HostName
HostName: not set
```
@ctorgalson allowing to unset the HostName is actually the point of this code change!

This is explained in this StackExchange thread:
https://apple.stackexchange.com/questions/30552/os-x-computer-name-not-matching-what-shows-on-terminal

"..It's better to NEVER set HostName, because doing so will make that value persist even when you later try to change your hostname via System Preferences. Conversely, the LocalHostName is picked up by the shell and hostname, so HostName is never needed."

please review that thread and consider NOT reverting this line of code.

Thanks for your interest

Reply:
> allowing to unset the HostName is actually the point of this code change

Aha! I thought there might be something more going on there. The change stays :)